ui/src/views/Tasks/submitTaskAction.js

```diff
@@ -13,7 +13,18 @@ import ajv from '../../utils/ajv';
 
 export default async ({ task, form, action, apolloClient, taskActions }) => {
   const actions = removeKeys(cloneDeep(taskActions), ['__typename']);
-  const taskGroup = task.taskId === task.taskGroupId ? task : task.decisionTask;
+  // We need to source scopes from somewhere. Typically these come from the task we're given,
+  // which is either a regular decision task or an action task. Decision tasks are easily
+  // identified, because their taskId matches the taskGroupId that they are in. Action tasks
+  // do not have such a mapping. Instead, we're simply going to try our best by falling back
+  // to `task` if `task.decisionTask` is null or undefined.
+  // At the time of writing, it's not clear whether the `task.decisionTask` case is even valid
+  // anymore, but it's being kept for the time being because verifying that is not a trivial
+  // matter.
```

Comment:
What is `task` when you trigger the action on a task group? The task group's UUID?

Reply:
good point. I think it should always be `task` object, checked code. Only used in places where `task` is guaranteed to be defined correctly

Reply:
It's a full task object. For example, when I use Cancel All on `V1ciHNV6S7eTWB3K1SakZA`, it ends up as:
```
__typename: "Task"
​​
created: "2024-03-20T14:10:57.718Z"
​​
deadline: "2024-03-21T14:10:57.718Z"
​​
dependencies: []
​​
extra: {…}
​​
metadata: {…}
​​​
__typename: "TaskMetadata"
​​​
description: "Initiate part or all of the training pipeline\n\nAction triggered by clientID `mozilla-auth0/ad|Mozilla-LDAP|bhearsum`\n"
​​​
name: "Action: Train"
​​​
owner: "action@noreply.mozilla.org"
​​​
source: "https://github.com/mozilla/firefox-translations-training/raw/dc91cb33531d7d20bcb5dcd4605381f8f2409999/.taskcluster.yml"
​​​
<prototype>: {…}
​​
payload: {…}
​​
priority: "LOWEST"
​​
requires: "ALL_COMPLETED"
​​
retries: 5
​​
routes: (3) […]
​​
schedulerId: "translations-level-1"
​​
scopes: (1) […]
​​​
0: "assume:repo:github.com/mozilla/firefox-translations-training:action:train"
​​​
length: 1
​​​
<prototype>: []
​​
tags: {…}
​​
taskGroupId: "WRffcusETWGVH66P9NI5yQ"
​​
taskId: "V1ciHNV6S7eTWB3K1SakZA"
​​
taskQueueId: "translations-1/decision-gcp"
​​
<prototype>: {…}
```

(Sorry for the bad paste...the Debugger doesn't let me copy the object nicely...)

Reply:
Oh duh, yeah that makes sense